lib/mocha/class_method.rb

```diff
@@ -53,7 +53,10 @@ def hide_original_method
     end
 
     def define_new_method
-      stub_method_owner.send(:define_method, method_name, stub_method_body(method_name))
+      self_in_scope = self
+      stub_method_owner.send(:define_method, method_name) do |*args, &block|
+        self_in_scope.instance.mocha.method_missing(self_in_scope.method_name, *args, &block)
+      end
```

Comment:
It looks as if this means you can remove the implementation of `ClassMethod#stub_method_body`, right?